src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_magmus.cpp

```diff
@@ -110,7 +117,61 @@ class boss_magmus : public CreatureScript
         }
 };
 
+class npc_ironhand_guardian : public CreatureScript
+{
+    public:
+
+        npc_ironhand_guardian() : CreatureScript("npc_ironhand_guardian") { }
+
+        struct npc_ironhand_guardianAI : public ScriptedAI
+        {
+
+            npc_ironhand_guardianAI(Creature* creature) : ScriptedAI(creature)
+            {
+                _instance = me->GetInstanceScript();
+                _lastState = 0;
+            }
+
+            void Reset() override
+            {
+                _events.Reset();
+            }
+
+            void UpdateAI(uint32 uiDiff) override
+            {
+                if (_lastState == 0)
```

Comment:
Missing UpdateVictim() call, this creature will never evade?

Reply:
I don't think so. This is a static non-selectable creature that will spew fire in the room. It is likely that implementing this fix can be done easier with UpdateVictim - if one is more knowledgeable in the TrinityCore internals.

Reply:
@martintopholm

Reply:
The Ironhand Guardian should not evade. When the party has entered the should keep casting Gout of Flame. Whether it should have a UpdateVictim anyways I am unsure about.